index.js

```diff
@@ -397,8 +395,8 @@ SendStream.prototype.isRangeFresh = function isRangeFresh(){
  */
 
 SendStream.prototype.redirect = function redirect(path) {
-  if (listenerCount(this, 'directory') !== 0) {
-    this.emit('directory')
+  if (listenerCount(this, 'directory')) {
+    this.emit('directory', path)
```

Comment:
Adding a new argument to the `directory` event does not seem related to adding an `index` event. Can you please restructure this pull request to only do what was described in the pul request summary and ping me when you have updated it?